core/src/main/scala/doric/syntax/Actions.scala

```diff
@@ -0,0 +1,56 @@
+package doric
+package syntax
+
+import cats.data.NonEmptyChain
+import cats.implicits.catsSyntaxValidatedId
+import doric.sem.{ColumnValidationError, Location}
+
+import org.apache.spark.sql.{Column, Dataset}
+
+trait Actions {
+  implicit class AcctionsOps[T](doric: DoricColumn[T]) {
+    def action: DoricActions[T] = DoricActions(doric)
+  }
+}
+
+case class DoricActions[T] private (doric: DoricColumn[T]) {
+  def validate: DoricValidations[T] = DoricValidations(doric)
+}
+
+case class DoricValidations[T] private (doric: DoricColumn[T]) {
+  def all(
```

Comment:
I think you should rename this to collect or something like that, because you are making a collect.
You should add some scaladocs too